project_task_project_required/models/project_task.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    is_project_required = fields.Boolean(
+        string='Is Project Required',
+        related='company_id.is_project_task_project_required',
+    )
+
+    @api.constrains('project_id')
+    def _check_project_id(self):
+        for task in self:
+            if task.is_project_required and not task.project_id:
```

Comment:
I don't know how can it happen in case?

Reply:
Odoo allows creating tasks and unsetting project_id after task has been created, so this check is the basically very essence of the module.

Reply:
My steps test
1.Go to Project -> All Task -> Create Task (don't depend project_id)
2.Go to Project -> Configuration -> check `Require Projects on Tasks`
3.Back to Task -> Project will required field.

Project is required field already and why are you want to check this function? or any case that used this function.

Reply:
That's marking UI field as required, it has nothing to do with function that ensures data integrity if I want to create task without project programmatically